sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -201,8 +201,26 @@ std::string GetHeadersAsString(Azure::Core::Http::Request const& request)
 void GetErrorAndThrow(const std::string& exceptionMessage)
 {
   DWORD error = GetLastError();
-  throw Azure::Core::Http::TransportException(
-      exceptionMessage + " Error Code: " + std::to_string(error) + ".");
+  char* errorString = nullptr;
+  std::string errorMessage = exceptionMessage + " Error Code: " + std::to_string(error);
+
+  if (FormatMessage(
+          FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_FROM_HMODULE | FORMAT_MESSAGE_ALLOCATE_BUFFER,
+          GetModuleHandle("winhttp.dll"),
+          error,
+          MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+          reinterpret_cast<LPSTR>(&errorString),
+          0,
+          nullptr)
+      != 0)
+  {
+    errorMessage += ": ";
+    errorMessage += errorString;
+    LocalFree(errorString);
```

Comment:
This is probably super defensive, but we can additionally guard from a memory leak in a rare case when nothing is right:
```suggestion
    try
    {
      errorMessage += ": ";
      errorMessage += errorString;
    }
    catch (...)
    {
      LocalFree(errorString);
      throw;
    }

    LocalFree(errorString);
```

Reply:
You're 100% right (and it's been annoying me), but I think I have a slightly better solution... Let me make the change and see if you like it.